units/commit_queue.go

```diff
@@ -284,10 +283,10 @@ func validatePR(pr *github.PullRequest) error {
 	if pr.GetMergeCommitSHA() == "" {
 		catcher.Add(errors.New("no merge commit SHA"))
 	}
-	if pr.GetUser() == nil || pr.GetUser().GetID() == 0 {
+	if pr.GetUser() == nil || pr.GetUser().GetLogin() == "" {
 		catcher.Add(errors.New("no valid user"))
 	}
-	if pr.GetBase() == nil || pr.GetBase().GetSHA() == "" {
+	if pr.GetBase() == nil || pr.GetBase().GetSHA() == "" || pr.GetBase().GetRef() == "" {
```

Comment:
does it make sense to make this a method on the pr type?

Reply:
The pr type is from go-github

Reply:
a helper function to reduce the complexity of the if clause then would help clean things up a bit and make the code a bit more self documenting

Reply:
done. I also relocated it to the thirdparty package.